src/client/datascience/jupyter/serverPreload.ts

```diff
@@ -59,13 +59,18 @@ export class ServerPreload implements IExtensionSingleActivationService {
             traceInfo(`Attempting to start a server because of preload conditions ...`);
 
             // Check if we are already connected
-            let providerConnection = await this.notebookProvider.connect({ getOnly: true, disableUI: true });
+            let providerConnection = await this.notebookProvider.connect({
+                getOnly: true,
+                disableUI: true,
+                resource: undefined
+            });
 
             // If it didn't start, attempt for local and if allowed.
             if (!providerConnection && !this.configService.getSettings(undefined).disableJupyterAutoStart) {
                 // Local case, try creating one
                 providerConnection = await this.notebookProvider.connect({
                     getOnly: false,
+                    resource: undefined,
```

Comment:
Does this then get updated later? Just making sure that if a server is preloaded that the undefined here was going to stay even when we start up the actual notebook later when opening the file.

Reply:
ChangeKernel should trigger the update it looks like?

Reply:
When we start the actual notebook, then we'd be connecting to some kernel, hence at that point with `Jupyter` we'd change the kernel.
And with Raw, we'd end up starting a different kernel.

Reply:
I.e. passing `undefined` is ok for auto start.